### ISSUE TITLE
Rule proposal: no-dynamic-require

### ISSUE THREAD

**Issue by axetroy** [sindresorhus/eslint-plugin-unicorn#2477]
### Description

The input to require should be a static string literal.

This is beneficial to various Bundles and Analyzer. And less difficult to migrate to es module.

The dynamic require should be prohibited.

### Fail

```js
require(pathToTargetModule)
require('./icons/' + iconType + '.svg')
```

### Pass

```js
require('./path/to/module')
require('./icons/pass.svg')
```

### Proposed rule name

no-dynamic-require

### Additional Info

Of course, in our project, there will indeed be such a situation.

```js
require(pathToTargetModule)
```

This is a typical scenario that needs to be ignored.

```
exports[`no-dynamic-require 1`] = `
## invalid(1): require(foo)

  > Input
  
        1 | require(foo)

  > Error 1/1: Unexpected dynamic require.
  
      > 1 | require(foo)
          | ^^^^^^^^^^^^

## invalid(2): require("foo" + "bar")

  > Input
  
        1 | require("foo" + "bar")

  > Error 1/1: Unexpected dynamic require.
  
      > 1 | require("foo" + "bar")
          | ^^^^^^^^^^^^^^^^^^^^^^

## invalid(3): require(\`\${foo}/bar\`)

  > Input
  
        1 | require(\`\${foo}/bar\`)

  > Error 1/1: Unexpected dynamic require.
  
      > 1 | require(\`\${foo}/bar\`)
          | ^^^^^^^^^^^^^^^^^^^^^
`;

```

**Comment by sindresorhus** [comment]
We are not adding any new CommonJS-related rules here.

And it seems more appropriate for https://github.com/un-ts/eslint-plugin-import-x anyway
